File: lib/model/timetable.dart
import 'dart:convert';

import 'package:intl/intl.dart';

class TimeTable {
  DateTime date;
  List<TimeTableItem> items;

  TimeTable(this.date, this.items);

  static TimeTable fromList(List<dynamic> list) {
    List<TimeTableItem> items = [];
    for (var item in list) {
      items.add(TimeTableItem(
          int.parse(item["PERIO"].toString()), item["ITRT_CNTNT"]));
    }

    return TimeTable(DateTime.now(), items);
  }

  String toJson() {
    List<Map<String, Object>> items = this
        .items
        .map((e) => {"period": e.period, "subject": e.subject})
        .toList();
    Map map = {
      "date": DateFormat("yyyy/MM/dd").format(this.date),
      "items": items
    };

    return jsonEncode(map);
  }
}

class TimeTableItem {
  int period;
  String subject;

  TimeTableItem(this.period, this.subject);
}

File: ios/Models/TimeTable.swift
//
//  TimeTable.swift
//  Runner
//
//  Created by 박성헌 on 2022/05/14.
//

import Foundation

struct TimeTable: Codable {
    let date: Date
    let items: [TimeTableItem]
    
    static func fromList(_ list: [[String : Any]]) -> TimeTable {
        let items: [TimeTableItem] = {
            var temp = [TimeTableItem]()
            for item in list {
                temp.append(TimeTableItem(period: Int((item["PERIO"] as! String))!, subject: (item["ITRT_CNTNT"] as! String)))
            }
            
            return temp
        }()
        return TimeTable(date: Date(), items: items)
    }
}

struct TimeTableItem: Codable {
    let period: Int
    let subject: String
}
